src/utils/context.ts

```diff
@@ -0,0 +1,247 @@
+ * @param node
+ * @return true if a dataset member, false otherwise
+ */
+export function isDsMember(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_MEMBER_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a dataset session
+ * @param node
+ * @return true if a dataset session, false otherwise
+ */
+export function isDsSession(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_SESSION_CONTEXT + ")").test(node.contextValue);
+}
+
+
+/**
+ * Helper function which identifies if the node is a partitioned dataset
+ * @param node
+ * @return true if a partitioned dataset, false otherwise
+ */
+export function isPds(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_PDS_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a USS Directory
+ * @param node
+ * @return true if a USS Directory, false otherwise
+ */
+export function isUssDirectory(node: TreeItem): boolean {
+    return new RegExp("^" + extension.USS_DIR_CONTEXT).test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a USS session
+ * @param node
+ * @return true if a USS session, false otherwise
+ */
+export function isUssSession(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.USS_SESSION_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a grouping or folder
+ * @param node
+ * @return true if a folder, false otherwise
+ */
+export function isFolder(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.JOBS_JOB_CONTEXT + "|" + extension.USS_DIR_CONTEXT + "|"
+                        + extension.DS_PDS_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a session
+ * @param node
+ * @return true if a session, false otherwise
+ */
+export function isSession(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.JOBS_SESSION_CONTEXT + "|" + extension.USS_SESSION_CONTEXT + "|"
+                        + extension.DS_SESSION_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a session but not a favorite
+ * @param node
+ * @return true if a session, false otherwise
+ */
+export function isSessionNotFav(node: TreeItem): boolean {
+    return new RegExp("^((?!.*" + extension.FAV_SUFFIX + ")(" + extension.JOBS_SESSION_CONTEXT + "|" + extension.USS_SESSION_CONTEXT + "|"
+                        + extension.DS_SESSION_CONTEXT + "))").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a session favorite
+ * @param node
+ * @return true if a session favorite, false otherwise
+ */
+export function isSessionFavorite(node: TreeItem): boolean {
+   // return isSession(node) && isFavorite(node);
+   return new RegExp("^(" + extension.FAVORITE_CONTEXT + ")").test(node.contextValue);
```

Comment:
If we only need to check if the node has Favorite Context can we remove this comment?

Reply:
Yes it is the root. Ironically I do have it removed in my follow up branch so will take it out now

Reply:
Comment removed